### PR TITLE
Add root gitconfig with autocrlf

### PR DESCRIPTION
**Pull request by rfay**
## The Problem:

Our users will use git from within the container, but there is no gitconfig, and as a result things like `git status` show problematic results wrt linefeeds.

This adds a gitconfig with  the very standard (for linux/mac autocrlf=input) (see https://help.github.com/articles/dealing-with-line-endings/#platform-linux)
```
[core]
	autocrlf = input
```

Long term it would be great if we weren't using root, which has its own problems. But here we are.

### CODE REVIEW

**Comment by rfay** [comment]
It's definitely not vague; I'm surprised the default was off, but when I exec-ed into the container git showed most of the files as having the wrong kind of line endings.

BTW, this will make people crazy on windows *if* they use CRLF, because it's the *wrong* setting there if they're working on a repo that has CRLF.

@beeradb did you test with a mounted /var/www/html without the /root/.gitconfig?

Bottom line is that we *should* have a /root/.gitconfig IMO, but if you have concerns about this being right we can revisit. Definitely worked for me.

**Comment by beeradb** [comment]
Here are my step-by-step test results:

- Start two sites running the different container versions 
Relevant `docker ps` output:
```
testing/oascratch ‹myproject/192-168-77-100:8443/system:admin/myproject› git:(master) » docker ps
CONTAINER ID        IMAGE                                         COMMAND                  CREATED             STATUS                             PORTS                                                                         NAMES
571170d6d4eb        drud/nginx-php-fpm7-local:v0.7.4-1-g9945bca   "/start.sh"              24 seconds ago      Up 23 seconds (healthy)            443/tcp, 0.0.0.0:32783->80/tcp, 0.0.0.0:32782->8025/tcp                       ddev-oascratch-web
4f39fcc03c44        drud/nginx-php-fpm7-local:v0.7.4              "/start.sh"              6 days ago          Up 3 minutes (healthy)             443/tcp, 0.0.0.0:32773->80/tcp, 0.0.0.0:32772->8025/tcp                       ddev-d7kickstart-web
```

The `v0.7.4-1-g9945bca` is the provisional tag for this PR.

- ddev SSH into the project using the tag from this PR (oascratch). Ensure the config is in place, and then run a `git status`

Results:
```
 root@571170d6d4eb:/var/www/html# cat /root/.gitconfig
[core]
	autocrlf = input
root@571170d6d4eb:/var/www/html# git status
On branch master
Your branch is up-to-date with 'origin/master'.
Untracked files:
  (use "git add <file>..." to include in what will be committed)

	.ddev/

nothing added to commit but untracked files present (use "git add" to track)
```

- ddev SSH into the project not using this PR. Run a ddev status for comparison:

```

testing/d7kickstart git:(master) » ddev ssh
root@4f39fcc03c44:/var/www/html# git status
On branch master
Your branch is up-to-date with 'origin/master'.
Untracked files:
  (use "git add <file>..." to include in what will be committed)

  .ddev/
  drush.settings.php

nothing added to commit but untracked files present (use "git add" to track)
```

**Comment by beeradb** [comment]
> Bottom line is that we should have a /root/.gitconfig IMO, but if you have concerns about this being right we can revisit. Definitely worked for me.

I don't actually have concerns. Just pointing out that it didn't seem to matter for me. I believe line endings are a real problem, though. I've seen it many places. Just can't seem to reproduce it here.

**Comment by rfay** [comment]
Here's an example repo that shows it very well, https://github.com/rfay/montclairdrive.org

```
root@855b7cad3f63:/var/www/html/docroot# git status
On branch site_d7
Your branch is up-to-date with 'origin/site_d7'.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   sites/all/libraries/ckeditor/CHANGES.md
	modified:   sites/all/libraries/ckeditor/LICENSE.md
	modified:   sites/all/libraries/ckeditor/build-config.js
	modified:   sites/all/libraries/ckeditor/contents.css
	modified:   sites/all/libraries/ckeditor/lang/af.js
	modified:   sites/all/libraries/ckeditor/lang/ar.js
	modified:   sites/all/libraries/ckeditor/lang/bg.js
	modified:   sites/all/libraries/ckeditor/lang/bn.js
	modified:   sites/all/libraries/ckeditor/lang/bs.js
	modified:   sites/all/libraries/ckeditor/lang/ca.js
	modified:   sites/all/libraries/ckeditor/lang/cs.js
	modified:   sites/all/libraries/ckeditor/lang/cy.js
	modified:   sites/all/libraries/ckeditor/lang/da.js
	modified:   sites/all/libraries/ckeditor/lang/de.js
	modified:   sites/all/libraries/ckeditor/lang/el.js
	modified:   sites/all/libraries/ckeditor/lang/en-au.js
	modified:   sites/all/libraries/ckeditor/lang/en-ca.js
	modified:   sites/all/libraries/ckeditor/lang/en-gb.js
	modified:   sites/all/libraries/ckeditor/lang/en.js
	modified:   sites/all/libraries/ckeditor/lang/eo.js
	modified:   sites/all/libraries/ckeditor/lang/es.js
	modified:   sites/all/libraries/ckeditor/lang/et.js
	modified:   sites/all/libraries/ckeditor/lang/eu.js
	modified:   sites/all/libraries/ckeditor/lang/fa.js
	modified:   sites/all/libraries/ckeditor/lang/fi.js
	modified:   sites/all/libraries/ckeditor/lang/fo.js
	modified:   sites/all/libraries/ckeditor/lang/fr-ca.js
	modified:   sites/all/libraries/ckeditor/lang/fr.js
	modified:   sites/all/libraries/ckeditor/lang/gl.js
	modified:   sites/all/libraries/ckeditor/lang/gu.js
	modified:   sites/all/libraries/ckeditor/lang/he.js
	modified:   sites/all/libraries/ckeditor/lang/hi.js
	modified:   sites/all/libraries/ckeditor/lang/hr.js
	modified:   sites/all/libraries/ckeditor/lang/hu.js
	modified:   sites/all/libraries/ckeditor/lang/id.js
	modified:   sites/all/libraries/ckeditor/lang/is.js
	modified:   sites/all/libraries/ckeditor/lang/it.js
	modified:   sites/all/libraries/ckeditor/lang/ja.js
	modified:   sites/all/libraries/ckeditor/lang/ka.js
	modified:   sites/all/libraries/ckeditor/lang/km.js
	modified:   sites/all/libraries/ckeditor/lang/ko.js
	modified:   sites/all/libraries/ckeditor/lang/ku.js
	modified:   sites/all/libraries/ckeditor/lang/lt.js
	modified:   sites/all/libraries/ckeditor/lang/lv.js
	modified:   sites/all/libraries/ckeditor/lang/mk.js
	modified:   sites/all/libraries/ckeditor/lang/mn.js
	modified:   sites/all/libraries/ckeditor/lang/ms.js
	modified:   sites/all/libraries/ckeditor/lang/nb.js
	modified:   sites/all/libraries/ckeditor/lang/nl.js
	modified:   sites/all/libraries/ckeditor/lang/no.js
	modified:   sites/all/libraries/ckeditor/lang/pl.js
	modified:   sites/all/libraries/ckeditor/lang/pt-br.js
	modified:   sites/all/libraries/ckeditor/lang/pt.js
	modified:   sites/all/libraries/ckeditor/lang/ro.js
	modified:   sites/all/libraries/ckeditor/lang/ru.js
	modified:   sites/all/libraries/ckeditor/lang/si.js
	modified:   sites/all/libraries/ckeditor/lang/sk.js
	modified:   sites/all/libraries/ckeditor/lang/sl.js
	modified:   sites/all/libraries/ckeditor/lang/sq.js
	modified:   sites/all/libraries/ckeditor/lang/sr-latn.js
	modified:   sites/all/libraries/ckeditor/lang/sr.js
	modified:   sites/all/libraries/ckeditor/lang/sv.js
	modified:   sites/all/libraries/ckeditor/lang/th.js
	modified:   sites/all/libraries/ckeditor/lang/tr.js
	modified:   sites/all/libraries/ckeditor/lang/tt.js
	modified:   sites/all/libraries/ckeditor/lang/ug.js
	modified:   sites/all/libraries/ckeditor/lang/uk.js
	modified:   sites/all/libraries/ckeditor/lang/vi.js
	modified:   sites/all/libraries/ckeditor/lang/zh-cn.js
	modified:   sites/all/libraries/ckeditor/lang/zh.js
	modified:   sites/all/libraries/ckeditor/plugins/a11yhelp/dialogs/lang/_translationstatus.txt
	modified:   sites/all/libraries/ckeditor/plugins/scayt/LICENSE.md
	modified:   sites/all/libraries/ckeditor/plugins/scayt/README.md
	modified:   sites/all/libraries/ckeditor/plugins/scayt/dialogs/toolbar.css
	modified:   sites/all/libraries/ckeditor/plugins/specialchar/dialogs/lang/_translationstatus.txt
	modified:   sites/all/libraries/ckeditor/plugins/wsc/LICENSE.md
	modified:   sites/all/libraries/ckeditor/plugins/wsc/README.md
	modified:   sites/all/libraries/ckeditor/plugins/wsc/dialogs/ciframe.html
	modified:   sites/all/libraries/ckeditor/plugins/wsc/dialogs/tmpFrameset.html
	modified:   sites/all/libraries/ckeditor/plugins/wsc/dialogs/wsc.css
	modified:   sites/all/libraries/ckeditor/skins/moono/dialog.css
	modified:   sites/all/libraries/ckeditor/skins/moono/dialog_ie.css
	modified:   sites/all/libraries/ckeditor/skins/moono/dialog_ie7.css
	modified:   sites/all/libraries/ckeditor/skins/moono/dialog_ie8.css
	modified:   sites/all/libraries/ckeditor/skins/moono/dialog_iequirks.css
	modified:   sites/all/libraries/ckeditor/skins/moono/editor.css
	modified:   sites/all/libraries/ckeditor/skins/moono/editor_gecko.css
	modified:   sites/all/libraries/ckeditor/skins/moono/editor_ie.css
	modified:   sites/all/libraries/ckeditor/skins/moono/editor_ie7.css
	modified:   sites/all/libraries/ckeditor/skins/moono/editor_ie8.css
	modified:   sites/all/libraries/ckeditor/skins/moono/editor_iequirks.css
	modified:   sites/all/libraries/ckeditor/skins/moono/readme.md
	modified:   sites/all/libraries/ckeditor/styles.js
	modified:   sites/all/modules/logintoboggan/INSTALL.txt
	modified:   sites/all/modules/logintoboggan/logintoboggan-rtl.css
```

If you do a `git config --global core.autocrlf input` (inside the container) everything straightens out.

As you can see, the problem is caused by actual files that were committed with CRLF in them, mostly cross-platform library type stuff, docroot/sites/all/libraries/ckeditor

I found 4-5 of my sites with this sort of issue, all mostly sites/all/libraries stuff where a cross-platform zipball is to be loaded there.

**Comment by rfay** [comment]
A more understandable resource on all this, explaining the issues per OS is at https://stackoverflow.com/questions/3206843/how-line-ending-conversions-work-with-git-core-autocrlf-between-different-operat - The bottom line is that core.autocrlf=input is probably what we want on there. core.autocrlf=input is normative on Macos and LInux. The default turns out to be "false", which is why this problem shows up.

**Comment by beeradb** [comment]
Nudge. any reason to not move forward with this?

**Comment by rfay** [comment]
Guess I lost track of that one!
